test/TESTING_GUIDELINES.md

```diff
@@ -1,85 +1,33 @@
+ 1. Run `test.sh` script OR
+ 2. Run: 
+  - `npm run build:dev`
+  - `npm run test:ui`
 
-## Maintenance
-
-When developing a new feature, in most cases you will perform changes only in places mentioned under [Test structure](#test-structure) section. If new flow is introduced, for example cross-device flow, a separate page object, step definition and `.feature` file can be created to handle the testing.
-
-If there's a need to introduce a change in `monster` gem, follow the steps:
-- create a branch out of master on `monster` gem
-- bump version in `monster/lib/monster/version.rb` according to your changes
-- for development purposes change `monster` gem dependency line in `test/Gemfile` of this project:
-  - if changes on monster branch: `gem 'monster', :git => 'git@bitbucket.org:onfido/monster.git', :branch => 'your-branch-on-monster'`
-  - if changes on local changes: `gem 'monster', :path => 'your/local/path/to/monster'`
-- run `bundle update monster` in `test/` directory
-- once solution tested, issue PR with your monster branch to `master` and assign 2 QAs
-- if PR approved, create tag with version from `version.rb`
-  - `git tag <version>` version
-- merge your `monster` branch to `master`
-- back in this project, update `monster` gem dependency line in `test/Gemfile` to reflect latest changes in monster by tag:
-  -  `gem 'monster', :git => 'git@bitbucket.org:onfido/monster.git', :tag => '<version_from_your_branch>'`
-- run `bundle update monster` in `test/` directory to apply the changes (`Gemfile.lock` should be updated)
-
-### Test structure
-
-All test artifacts are in `features/` directory:
-- `page_objects/` - contains files with page objects that map objects i.e by id or css
-- `step_definitions/` - contains generic, commonly shared steps used in `.feature` files
-- `*.feature` - behavior-driven tests written in Gherkin
-- `support/` - contains support for setting up tests, i.e requires `monster` gem dependency
-- `scripts/` - contains scripts to setup test environment on Travis CI
+### `test` directory structure
 
+- `pageobjects/`contains files with page objects that map objects by css selectors and functions that use these page objects
+- `resources/`contains files needed for the upload tests
+- `scripts/` contains scripts to setup test environment on Travis CI
+- `specs/` contains files with the tests implemented
+- `utils/` contains util files with the set up for accessibility testing, browserstack and mocha
+- `config.json`contains configuration for the browsers on which the tests will run
 
 ### Troubleshooting
-
 If you face issues regarding test setup or weird errors during test execution, there are some things to check:
+
 - make sure your Google Chrome is up to date
 - make sure your Chromedriver is up to date
-- make sure webpack server starts properly (`npm run dev`)
+- make sure your webserver is up and running (`npm run travis`)
```

Comment:
What's the difference between `npm run dev` and `npm run travis`?

Reply:
Not sure what was the difference but currently even if I run this command on `development` I get

npm run dev

```
> onfido-sdk-ui@5.2.3 dev /Users/bart.ziemba/Projects/onfido-sdk-ui
> NODE_ENV=development webpack-dev-server --inline --hot --progress --https

 10% [1] building 1/1 modules 0 active(node:32667) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
events.js:173
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE: address already in use 0.0.0.0:8080
    at Server.setupListenHandle [as _listen2] (net.js:1255:14)
    at listenInCluster (net.js:1303:12)
    at doListen (net.js:1442:7)
    at internalTickCallback (internal/process/next_tick.js:72:19)
    at process._tickCallback (internal/process/next_tick.js:47:5)
    at Function.Module.runMain (internal/modules/cjs/loader.js:777:11)
    at executeUserCode (internal/bootstrap/node.js:342:17)
    at startExecution (internal/bootstrap/node.js:276:5)
    at startup (internal/bootstrap/node.js:227:5)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:743:3)
Emitted 'error' event at:
    at emitErrorNT (net.js:1282:8)
    at internalTickCallback (internal/process/next_tick.js:72:19)
    [... lines matching original stack trace ...]
    at bootstrapNodeJSCore (internal/bootstrap/node.js:743:3)
```
It seems this command didn't work for a while.

Reply:
That's the error I get locally if I run `npm run travis` first, leave it running, then go run `npm run dev` in another Terminal window. 

Not a huge issue but the Travis server command `npm run travis` knows to try to start the local Travis server on another port (8081 usually) while the Webpack dev-server that is part of `npm run dev` doesn't. So the order they are run if you want to run both makes a difference.

Reply:
`npm run dev` works. The reason why it doesn't work on your machine is because that port is already in use. To kill it just run `sudo lsof -i tcp:8080`

Reply:
OK, thanks for clarifications, so in that case, it probably makes sense to keep `npm run dev` in the guidelines?

Reply:
Yeah, though it might need to be rephrased to explain that if running `npm run dev` gives you an error that means that the local Travis server is already running on port 8080. 

BTW, I just checked and if you do run `npm run travis` if it is already running, that will just start up Travis server on another port while existing running one keeps going.

Reply:
Done 1ba3c55